app/services/decisions/simple_decision_tree.rb

```diff
@@ -1,150 +1,64 @@
-        add_view: :work_item,
-        sub_id: :work_item_id,
-        form: Steps::WorkItemForm,
-        proceed_url: edit(:letters_calls),
-      )
-    end
-
-    def after_disbursements
-      add_another(
-        scope: application.disbursements,
-        add_view: :disbursement_type,
-        sub_id: :disbursement_id,
-        form: [Steps::DisbursementTypeForm, Steps::DisbursementCostForm],
-        proceed_url: show(:cost_summary),
-      )
-    end
-
-    def after_firm_details
-      create_new_or_summary(
-        page: :defendant_details,
-        summary_page: :defendant_summary,
-        nested_id: :defendant_id,
-        scope: application.defendants,
-      )
-    end
-
-    def after_claim_details
-      create_new_or_summary(
-        page: :work_item,
-        summary_page: :work_items,
-        nested_id: :work_item_id,
-        scope: application.work_items,
-      )
-    end
-
-    def after_work_item_delete
-      create_new_or_summary(
-        page: :work_item,
-        summary_page: :work_items,
-        nested_id: :work_item_id,
-        scope: application.work_items,
-      )
-    end
-
-    def after_disbursement_delete
-      create_new_or_summary(
-        page: :disbursement_type,
-        summary_page: :disbursements,
-        nested_id: :disbursement_id,
-        scope: application.disbursements,
-      )
-    end
-
-    def after_letters_calls
-      create_new_or_summary(
-        page: :disbursement_type,
-        summary_page: :disbursements,
-        nested_id: :disbursement_id,
-        options: { edit_when_one: true },
-        scope: application.disbursements
-      )
-    end
-
-    def add_another(scope:, add_view:, sub_id:, form:, proceed_url:)
-      if form_object.add_another.yes?
-        edit(add_view, sub_id => StartPage::NEW_RECORD)
-      else
-        # we direct the user to any invalid forms when they choose next
-        forms = Array(form)
-        invalid_instance = scope.detect { |record| forms.any? { |f| !f.build(record, application:).valid? } }
-        if invalid_instance
-          edit(add_view, sub_id => invalid_instance.id, :flash => { error: 'Can not continue until valid!' })
-        else
-          proceed_url
-        end
-      end
-    end
-
-    def create_new_or_summary(page:, summary_page:, nested_id:, scope:, options: { edit_when_one: false })
-      count = scope.count
-      if count.zero?
-        edit(page, nested_id => StartPage::NEW_RECORD)
-      elsif count == 1 && options[:edit_when_one]
-        new_work_item = scope.first
-        edit(page, nested_id => new_work_item.id)
-      else
-        edit(summary_page)
-      end
-    end
+    from(:other_info).goto(edit: :equality)
+    from(:equality)
+      # .when(-> { answer_equality.yes? }).goto(edit: :equality_answers)
+      .when(-> { answer_equality.yes? }).goto(show: :start_page, answer: 'yes')
+      .goto(edit: :solicitor_declaration)
+    from(:solicitor_declaration).goto(show: :claim_confirmation)
```

Comment:
I know its a lot, but are we planning to include any test around this ?

Reply:
So the components have been tested.. as for the actual implementation that is tested by the existing tests... as this is a refactor none of them chnaged and we still have 100% coverage. :)